### PR TITLE
Lwang

### PR DESCRIPTION
**Pull request by lambertwang-zz**

### CODE REVIEW

**Comment by magellantoo** [comment]
Merged

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,10 @@
+
 old_index.html
 gallerydata/
 config.toml
+
+# vs code settings
+.vscode/
+
+# Binaries
+server.exe
```
Comment (tmathmeyer) [inline .gitignore:9]: server development on windows......... what (also get a real editor)

File: readme.md [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
- 1 copy config.toml.example to config.toml  
- 2 put your gmaps api key into config.toml  
- 3 make a directory called 'gallerydata'  
- 4 in hike data make a directory for each album  
- 5 in each album make two directories: 'img', 'pan'  
- 6 put images in to the img dir and panoramics into the pan dir  
- 7 run `go build server.go`  
- 8 run `server`  
+## Setup
+1. Copy 'config.toml.example' to 'config.toml'
```
Comment (tmathmeyer) [inline readme.md]: uncapitolize. too much grammar (also use ` instead of ')

File: readme.md [hunk 3]
```diff
@@ -1,21 +1,21 @@
...
- 4 in hike data make a directory for each album  
- 5 in each album make two directories: 'img', 'pan'  
- 6 put images in to the img dir and panoramics into the pan dir  
- 7 run `go build server.go`  
- 8 run `server`  
+## Setup
+1. Copy 'config.toml.example' to 'config.toml'  
+2. put your gmaps api key into config.toml  
+3. make a directory called 'gallerydata/'  
+4. in 'gallerydata/' make a directory for each album
```
Comment (tmathmeyer) [inline readme.md]: ` not '

File: server.go [hunk 4]
```diff
@@ -2,79 +2,90 @@ package main
...
+
+	"github.com/BurntSushi/toml"
 )
 
+const serverPort string = ":8081"
+const galleryDataDir string = "./gallerydata/"
+const galleryDetail string = "./gallerydetail/"
+const thumbPrefix string = "tn_"
+
+// Hike stores hike information for displaying on index page
```
Comment (tmathmeyer) [inline server.go]: no comments

File: server.go [hunk 5]
```diff
@@ -2,79 +2,90 @@ package main
...
 type Hike struct {
 	HikeName string
 	HikeID   string
 	HikeLat  string
 	HikeLng  string
 }
 
+// Config stores server configuration
```
Comment (tmathmeyer) [inline server.go]: no comments

File: server.go [hunk 6]
```diff
@@ -2,79 +2,90 @@ package main
...
 type Config struct {
 	Title  string
-	ApiKey string
+	APIKey string
```
Comment (tmathmeyer) [inline server.go]: Camel case on the fields (even if API is an acronym)

File: server.go [hunk 7]
```diff
@@ -2,79 +2,90 @@ package main
...
 	Hikes  []Hike
 }
 
+// Image stores locator and description
 type Image struct {
-	Uri  string
+	URI  string
```
Comment (tmathmeyer) [inline server.go]: CamelCase

File: server.go [hunk 8]
```diff
@@ -2,79 +2,90 @@ package main
...
 	Desc string
 }
 
+// HikeDetail stores hike information for displaying on contents page
```
Comment (tmathmeyer) [inline server.go]: no comments

File: server.go [hunk 9]
```diff
@@ -2,79 +2,90 @@ package main
...
 	Panos    []Image
 }
 
-func hikeDetailhandler(w http.ResponseWriter, r *http.Request) {
+func hikeDetailhandler(w http.ResponseWriter, req *http.Request) {
 	var config Config
 	if _, err := toml.DecodeFile("config.toml", &config); err != nil {
 		fmt.Printf("%s\n", err)
-		http.NotFound(w, r)
+		http.NotFound(w, req)
```
Comment (tmathmeyer) [inline server.go]: add a return too

File: server.go [hunk 10]
```diff
@@ -2,79 +2,90 @@ package main
...
+// HikeDetail stores hike information for displaying on contents page
 type HikeDetail struct {
 	HikeName string
 	HikeID   string
 	Images   []Image
 	Panos    []Image
 }
 
-func hikeDetailhandler(w http.ResponseWriter, r *http.Request) {
+func hikeDetailhandler(w http.ResponseWriter, req *http.Request) {
```
Comment (tmathmeyer) [inline server.go]: please keep this as r (standard naming for this)

File: server.go [hunk 11]
```diff
@@ -2,79 +2,90 @@ package main
...
-	"github.com/BurntSushi/toml"
 	"html/template"
 	"io/ioutil"
 	"net/http"
 	"os"
 	"os/exec"
+	"path/filepath"
 	"strings"
+
+	"github.com/BurntSushi/toml"
```
Comment (tmathmeyer) [inline server.go:13]: dont break out, also packages go alphabetically

File: server.go [hunk 12]
```diff
@@ -90,61 +101,73 @@ func hikeDetailhandler(w http.ResponseWriter, r *http.Request) {
...
 	}
-	http.NotFound(w, r)
+	http.NotFound(w, req)
 }
 
-func imageHandler(w http.ResponseWriter, r *http.Request) {
-	typepath := strings.SplitN(r.URL.Path[len("/img/"):], "/", 3)
-	if len(typepath) != 3 {
-		http.NotFound(w, r)
+func imageHandler(w http.ResponseWriter, req *http.Request) {
```
Comment (tmathmeyer) [inline server.go]: keep as r plz

File: server.go [hunk 13]
```diff
@@ -90,61 +101,73 @@ func hikeDetailhandler(w http.ResponseWriter, r *http.Request) {
...
+	// Path expected: /img/[[size]]/[[HikeId]]/[[ImageName]]
```
Comment (tmathmeyer) [inline server.go]: no comments

File: server.go [hunk 14]
```diff
@@ -90,61 +101,73 @@ func hikeDetailhandler(w http.ResponseWriter, r *http.Request) {
...
+	if imgSize == "full" {
+		http.ServeFile(w, req, imgPath)
 		return
 	}
 
-	if typepath[0] == "small" {
-		img_path := "./gallerydata/" + typepath[1] + "/img/" + typepath[2]
-		thm_path := "./gallerydata/" + typepath[1] + "/img/tn_" + typepath[2]
+	if imgSize == "small" {
+		// Jank code to replace file extensions
```
Comment (tmathmeyer) [inline server.go]: comment

File: server.go [hunk 15]
```diff
@@ -90,61 +101,73 @@ func hikeDetailhandler(w http.ResponseWriter, r *http.Request) {
...
+		thmPath := galleryDataDir + hikeID + imgDir + thumbPrefix + strings.TrimSuffix(imgName, filepath.Ext(imgName)) + ".jpg"
 
-		if !exists(thm_path) {
-			cmd := exec.Command("./vipsthumbnail", "-s", "450", img_path)
+		if !exists(thmPath) {
+			// vipsthumbnail will only produce .jpg images
+			fmt.Printf("Creating thumbnail for Image: %s\n", thmPath)
```
Comment (tmathmeyer) [inline server.go]: why print this?

File: server.go [hunk 16]
```diff
@@ -90,61 +101,73 @@ func hikeDetailhandler(w http.ResponseWriter, r *http.Request) {
...
+		}
 	}
 
-	if typepath[0] == "full" {
-		img_path := "./gallerydata/" + typepath[1] + "/img/" + typepath[2]
-		http.ServeFile(w, r, img_path)
+	// ie img/ or pan/
+	imgDir := typepath[0] + "/"
+	imgSize := typepath[1]
+	hikeID := typepath[2] + "/"
```
Comment (tmathmeyer) [inline server.go]: might as well change to galleryID

File: static/detail.css [hunk 17]
```diff
@@ -0,0 +1,54 @@
+
```
Comment (tmathmeyer) [inline static/detail.css]: remove line

File: server.go (hunk 11 above)
Comment (lambertwang-zz) [reply server.go:13]: This is what gofmt and golint says you should do though. 
https://github.com/golang/go/wiki/CodeReviewComments#imports

File: server.go (hunk 6 above)
Comment (lambertwang-zz) [reply server.go]: https://github.com/golang/go/wiki/CodeReviewComments#initialisms

File: server.go (hunk 11 above)
Comment (tmathmeyer) [reply server.go:13]: sounds good

**Pull request merged**